MetadataExtractor/StringValue.cs

```diff
@@ -87,7 +87,23 @@ int IConvertible.ToInt32(IFormatProvider provider)
 
         ushort IConvertible.ToUInt16(IFormatProvider provider) => ushort.Parse(ToString());
 
-        uint IConvertible.ToUInt32(IFormatProvider provider) => uint.Parse(ToString());
+        uint IConvertible.ToUInt32(IFormatProvider provider)
+        {
+            try
+            {
+                return uint.Parse(ToString());
+            }
+            catch(Exception)
+            {
+                ulong val = 0;
+                foreach (var b in Bytes)
+                {
+                    val = val << 8;
+                    val += b;
```

Comment:
I'm not this approach makes sense if there are more than four bytes (or perhaps also when fewer).